content/terraform/v1.12.x/docs/language/meta-arguments.mdx

```diff
@@ -0,0 +1,211 @@
+
+- [Manage similar resources with for each](/terraform/tutorials/configuration-language/for-each)
+
+### Examples
+
+- [Create multiple ephemeral resources](/terraform/language/block/ephemeral#create-multiple-ephemeral-resources)
+- [Import multiple existing resources](terraform/language/block/import#import-multiple-resources)
+- [Create multiple instances of module resources](/terraform/language/block/module#create-multiple-instances-of-module-resources) 
+- [Create multiple instances of a resource](/terraform/language/block/resource#create-multiple-instances-of-a-resource) 
+
+### Guidance 
+
+- [Creat multiple instances](/terraform/language/data-sources#multiple-instances) when querying a data source
+- [Create multiple instances](/terraform/language/modules#create-multiple-copies-of-modules) when calling modules
+
+### References
+
+- [`for_each` argument in `data` blocks](/terraform/language/block/data#for_each) 
+- [`for_each` argument in `ephemeral` blocks](/terraform/language/block/ephemeral#for_each)
+- [`for_each` argument in `module` blocks](/terraform/language/block/module#for_each)
+- [`for_each` argument in. `resource` blocks](/terraform/language/block/resource#for_each)
+
+## `provider` 
+
+By default, Terraform determines the local name of the provider from the first word in the resource type and uses that provider's default configuration to create the resource. For example, the resource type `google_compute_instance` automatically uses the default configuration for the `google` provider .
+
+You can add multiple `provider` blocks to your configuration and use the `provider` argument to a resource definition to specify which provider it should use. Creating multiple configurations for a single provider is a common pattern for managing resources in different regions of multi-region services.
+
+The `provider` argument is distinct from the `provider` block, which contains provider configuration settings. Refer to [Providers](/terraform/language/providers) for more information about the `provider` block.
+
+### Examples
+
+- [Specify an alternate provider configuration](/terraform/language/block/data#select-an-alternate-provider-configuration) for a data source
+- [Specify an alternate provider configuration](/terraform/language/block/ephemeral#use-alternate-provider-configurations) for an ephemeral resource
+- [Select an alternate provider configuration](/terraform/language/block/import#select-an-alternate-provider-configuration) for imported resources
+- [Select a provider without a default configuration](/terraform/language/block/provider#provider-without-a-default-configuration) 
+- [Select an alternative provider configuration](/terraform/language/block/resource#select-an-alternate-provider-configuration) for resources
+
+### Guidance
+
+- [Providers](/terraform/language/providers)
+- [Select an alternate provider configuration](/terraform/language/data-sources#alternate-provider-configurations) when querying a data source
```

Comment:
I feel like maybe we should just have the provider page here? Since the data source example is above?